contracts/liquidity_hub/vault-network/vault_factory/src/execute/migrate_vaults.rs

```diff
@@ -0,0 +1,241 @@
+use cosmwasm_std::{to_binary, Addr, CosmosMsg, DepsMut, MessageInfo, Response, WasmMsg};
+
+use vault_network::vault::MigrateMsg;
+
+use crate::state::read_vaults;
+use crate::{
+    err::{StdResult, VaultFactoryError},
+    state::CONFIG,
+};
+
+pub fn migrate_vaults(
+    deps: DepsMut,
+    info: MessageInfo,
+    vault_addr: Option<String>,
+    vault_code_id: u64,
+) -> StdResult<Response> {
+    // check that owner is creating vault
+    let config = CONFIG.load(deps.storage)?;
+    if config.owner != info.sender {
+        return Err(VaultFactoryError::Unauthorized {});
+    }
+
+    // migrate only the provided vault address, otherwise migrate all vaults
+    let mut res = Response::new().add_attributes(vec![
+        ("method", "migrate_vaults".to_string()),
+        ("code_id", vault_code_id.to_string()),
+    ]);
+    return if let Some(vault_addr) = vault_addr {
+        Ok(res
+            .add_attribute("vault", vault_addr.clone())
+            .add_message(migrate_vault_msg(
+                deps.api.addr_validate(vault_addr.as_str())?,
+                vault_code_id,
+            )?))
+    } else {
+        let vaults = read_vaults(deps.storage, deps.api, None, Some(30u32))?;
+        for vault in vaults {
+            res = res
```

Comment:
are you sure you need to do `res = res`? Is the initial type not modified anyways so you can just refer to `res` below

Reply:
mmm good catch, I was not sure about it, will try it with the test :D

Reply:
No doesn't work, `res` is moved in the for loop